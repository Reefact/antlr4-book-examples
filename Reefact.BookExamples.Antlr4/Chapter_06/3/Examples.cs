#region Usings declarations

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_06._3 {

    [UseReporter(typeof(VisualStudioReporter))]
    public class Examples {

        [Fact]
        [GraphicalTree("ParseTree1.svg")]
        public void tokens_tree() {
            // Setup
            GRun grun = GRun.ReadResource("t.dot", 6, 3);
            // Exercise
            string parseTree1 = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(parseTree1);
        }

    }

}