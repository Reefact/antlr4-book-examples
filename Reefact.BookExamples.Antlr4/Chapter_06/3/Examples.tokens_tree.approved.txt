graph TD
	1["graph"] --> 2["digraph"]
	1 --> 3["id"]
	3 --> 4["G"]
	1 --> 5["{"]
	1 --> 6["stmt_list"]
	6 --> 7["stmt"]
	7 --> 8["id"]
	8 --> 9["rankdir"]
	7 --> 10["="]
	7 --> 11["id"]
	11 --> 12["LR"]
	6 --> 13[";"]
	6 --> 14["stmt"]
	14 --> 15["node_stmt"]
	15 --> 16["node_id"]
	16 --> 17["id"]
	17 --> 18["main"]
	15 --> 19["attr_list"]
	19 --> 20["["]
	19 --> 21["a_list"]
	21 --> 22["id"]
	22 --> 23["shape"]
	21 --> 24["="]
	21 --> 25["id"]
	25 --> 26["box"]
	19 --> 27["]"]
	6 --> 28[";"]
	6 --> 29["stmt"]
	29 --> 30["edge_stmt"]
	30 --> 31["node_id"]
	31 --> 32["id"]
	32 --> 33["main"]
	30 --> 34["edgeRHS"]
	34 --> 35["edgeop"]
	35 --> 36["-&gt;"]
	34 --> 37["node_id"]
	37 --> 38["id"]
	38 --> 39["f"]
	34 --> 40["edgeop"]
	40 --> 41["-&gt;"]
	34 --> 42["node_id"]
	42 --> 43["id"]
	43 --> 44["g"]
	6 --> 45[";"]
	6 --> 46["stmt"]
	46 --> 47["edge_stmt"]
	47 --> 48["node_id"]
	48 --> 49["id"]
	49 --> 50["f"]
	47 --> 51["edgeRHS"]
	51 --> 52["edgeop"]
	52 --> 53["-&gt;"]
	51 --> 54["node_id"]
	54 --> 55["id"]
	55 --> 56["f"]
	47 --> 57["attr_list"]
	57 --> 58["["]
	57 --> 59["a_list"]
	59 --> 60["id"]
	60 --> 61["style"]
	59 --> 62["="]
	59 --> 63["id"]
	63 --> 64["dotted"]
	57 --> 65["]"]
	6 --> 66[";"]
	6 --> 67["stmt"]
	67 --> 68["edge_stmt"]
	68 --> 69["node_id"]
	69 --> 70["id"]
	70 --> 71["f"]
	68 --> 72["edgeRHS"]
	72 --> 73["edgeop"]
	73 --> 74["-&gt;"]
	72 --> 75["node_id"]
	75 --> 76["id"]
	76 --> 77["h"]
	6 --> 78[";"]
	1 --> 79["}"]
